### ISSUE TITLE
variational inference: global step not exposed to custom optimizers

### ISSUE THREAD

**Issue by mariru** [blei-lab/edward#635]
By default, the variational inference class uses ADAM with decaying rate as optimizer.
In large models with many parameters using ADAM might be too costly, and using a basic gradient descent optimizer is more appropriate.
However, other optimizers can't use a decaying learning rate because `global_step` is not exposed.

A slight API change would be nice. 
One option is a flag that can be passed to  `inference.initialize`, like 'default_optimizer_adaptive' which by default is False. When it's True, the optimizer would default to the tf.GradientDescentOptimizer with decaying learning rate.

Alternatively one could add the option to pass `optimizer='gradientdescent_with_decay'` to `inference.initialize()`. 

Thoughts?

**Comment by dustinvtran** [comment]
That makes sense. To keep the naming in line with TensorFlow, we could just call the argument `global_step` and which can be True or False.

**Comment by kashif** [comment]
@dustinvtran there is already a var `global_step` in the function. Do you propose to rename it?

**Comment by dustinvtran** [comment]
Good point. It would be bad if the argument is a bool and yet the internal TensorFlow function uses a `tf.Variable`. To keep things most general, I guess we could make the argument `global_step=None`, and the user can pass in a `tf.Variable`. This way `global_step` can be used across multiple inferences if desired(?). Other suggestions are welcome.

**Comment by kashif** [comment]
cool... will check

**Comment by kashif** [comment]
can be closed
